pkg/network/setup/netstat.go

```diff
@@ -114,45 +114,64 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
 
 	primaryNetwork := netvmispec.LookupPodNetwork(vmi.Spec.Networks)
-	interfacesStatus = restorePodIfaceNameForPrimaryIface(interfacesStatus, primaryNetwork, vmi.Status.Interfaces)
+	if primaryNetwork != nil {
+		interfacesStatus = restorePrimaryIfaceStatus(interfacesStatus, vmi.Status.Interfaces, primaryNetwork.Name)
+	}
 	primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, primaryNetwork)
 	if primaryInterfaceStatus != nil {
 		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
 	}
 
 	interfacesStatus = ifacesStatusFromMultus(interfacesStatus, multusStatusNetworksByName, vmiInterfacesSpecByName)
 
+	interfacesStatus = restorePodIfaceNames(interfacesStatus, vmi.Status.Interfaces)
```

Comment:
The `NetStat.UpdateStatus` method builds the VMI.Status.Interfaces slice in stages.
Each stage enriches the previous stage - it could modify existing elements or add new ones.
When a slice grows, its underlying array could be replaced - thus there is a need to assign `interfacesStatus` multiple times.
For additional details, please see https://go.dev/blog/slices-intro and https://go.dev/ref/spec#Appending_and_copying_slices.

Reply:
> I'm OK with your response in the context of this PR, and you can consider it resolved. I am not however OK with the was this function works, in so called "layers". I am suggesting to refactor this method in a future PR and design it in a "map/reduce" style. It collects information about interfaces from multiple sources, and in each step it both collects, and decides what to use and what to override. I am suggesting to split it into 2 separate steps: first collect all the information from all sources into a single "table" in memory, and then as a separate step, analyze the collected data and yield a consensus about which data is "right" per interface

I would be happy to review your PR if you would like to propose a refactoring of the `UpdateStatus` method.